### DISCUSSION TITLE
Is there a way to run a server-side command (or a PHP/AJAX function) when 504 error occurs?

### DISCUSSION THREAD

**Discussion by tukusejssirs** [Studio-42/elFinder#3341] [Q&A]
@nao-pon, I need to manage many SMB shares (from Windows mounted on Linux server) for many users via a local website (all shares are available for all users), therefore I opted to mount the shares directly on the server and then add them as simple local storage in elFinder. All those shares should be permanent unless the remote (Windows) is down.

The only issue I currently experience is that when the machine goes offline, it tooks elFinder several seconds (10+ seconds) to notice this and then the following error shows up:

```php
Invalid backend response.
Error occurred on the server side.
HTTP error 504
```

Is there a way to run a command (or a PHP/AJAX function) on the server when this error occurs? Then I’d destroy the elFinder instance and create a new one.

Thanks for the help!

**Comment by mguinness** [comment]
A quick and dirty way would be to use jQuery [.ajaxError()](https://api.jquery.com/ajaxerror/) to catch 504 Gateway Timeout.

**Comment by tukusejssirs** [comment]
Thanks for the answer @mguinness, I still need to find some time to try and implement your suggestion. That said it looks promising. :wink:

---

Here are some (more or less) related issues:
- #2351 (`504` error)
- #2328 (loading metadata of many files is quite slow and outputs `504`);
- #1908 (creating a large archive is very slow and outputs `504`);
- #3282 (long query string).

---

A partially OT questions: is there a way to make elFinder change the `max_execution_time` (generally: allow unlimited execution time for a particular process/function/whatever) when needed? I wouldn’t like to set it globally, just when needed (I already to this with other pages in a website where I use elFinder, but this does not work for elFinder for me for some reason).

---

**Update:** Could elFinder query files and their metadata async’ly? That is as soon the metadata of a file are gathered, the file is shown in the UI. Also when the execution time limit is approaching, elFinder should start the gathering again (of course not quering files of which we already gathered the metadata).

**Update 2:** Can elFinder (or I) use [`set-time-limit`](https://www.php.net/manual/en/function.set-time-limit.php) together with checking if it is near the time limit, when it should be reset?

**Reply by nao-pon** [reply]
There are the following items as options to know the changes on the file system.

https://github.com/Studio-42/elFinder/blob/1b9707cb97f5a328aeaec52cefbe9ff1a4c0281d/php/elFinderVolumeDriver.class.php#L468-L486

**Comment by nao-pon** [comment]
@tukusejssirs You can extend the execution time by calling `elFinder::checkAborted();` on PHP. If you are iterating, you can call it appropriately to prevent timeouts.

However, be aware that PHP does not control timeouts other than max_execution_time for Nginx servers.
